addons/point_of_sale/migrations/8.0.1.0.1/post-migration.py

```diff
@@ -0,0 +1,210 @@
+    """
+    pt_obj = pool['stock.picking.type']
+    wh = pool['stock.warehouse'].browse(
+        cr, SUPERUSER_ID, get_warehouse_id(cr, pool, shop_id))
+    return pt_obj.create(cr, SUPERUSER_ID, {
+        'name': pc.name,
+        'sequence_id': pool['ir.model.data'].xmlid_to_res_id(
+            cr, SUPERUSER_ID, 'point_of_sale.seq_picking_type_posout'),
+        'code': 'outgoing',
+        'warehouse_id': wh.id,
+        'default_location_src_id': wh.wh_output_stock_loc_id.id,
+        'default_location_dest_id': pool['ir.model.data'].xmlid_to_res_id(
+            cr, SUPERUSER_ID, 'stock.stock_location_customers'),
+    })
+
+
+def migrate_pos_config(cr, pool):
+    pc_obj = pool['pos.config']
+    pc_ids = pc_obj.search(cr, SUPERUSER_ID, [])
+    for pc in pc_obj.browse(cr, SUPERUSER_ID, pc_ids):
+        cr.execute("""
+            SELECT %s
+            FROM pos_config
+            WHERE id = %d
+        """ % (openupgrade.get_legacy_name('shop_id'), pc.id))
+        shop_id = cr.fetchone()[0]
+        stock_loc_id = get_stock_location_id(cr, pool, shop_id)
+        vals = {
+            'stock_location_id': stock_loc_id,
+            'company_id': get_company_id(cr, pool, shop_id, pc, stock_loc_id),
+            'picking_type_id': get_picking_type_id(cr, pool, shop_id, pc),
+        }
+        pricelist_id = get_pricelist_id(cr, pool, shop_id, pc)
+        if pricelist_id:
+            vals.update({'pricelist_id': pricelist_id})
+        pc.write(vals)
+
+
+def available_in_pos_field_func(cr, pool, id, vals):
+    logger.warning(
+        'available_in_pos of product.template %d has been set to True '
+        'whereas at least one of its product_product was False', id)
+    return any(vals)
+
+
+def expense_pdt_field_func(cr, pool, id, vals):
+    logger.warning(
+        'expense_pdt of product.template %d has been set to True '
+        'whereas at least one of its product_product was False', id)
+    return any(vals)
+
+
+def income_pdt_field_func(cr, pool, id, vals):
+    logger.warning(
+        'income_pdt of product.template %d has been set to True '
+        'whereas at least one of its product_product was False', id)
+    return any(vals)
+
+
+def to_weight_field_func(cr, pool, id, vals):
+    logger.warning(
+        'to_weight of product.template %d has been set to True '
+        'whereas at least one of its product_product was False', id)
+    return any(vals)
+
+
+def set_proxy_ip(cr, pool):
+    pc_obj = pool['pos.config']
+    pc_ids = pc_obj.search(cr, SUPERUSER_ID, [])
+    for pc in pc_obj.browse(cr, SUPERUSER_ID, pc_ids):
+        if (
+            pc.iface_cashdrawer or pc.iface_payment_terminal
+            or pc.iface_electronic_scale or pc.iface_print_via_proxy
+        ):
+            pc.write({'proxy_ip': 'http://localhost:8069'})
+
+
+@openupgrade.migrate()
+def migrate(cr, version):
+    pool = RegistryManager.get(cr.dbname)
+    openupgrade.move_field_m2o(
```

Comment:
You can encapsulate these calls to `move_field_m2o` for clarity in a method. You can even iterate through a list of the fields and make the call to the method with that variable as argument.